query-engine/connectors/query-connector/src/write_args.rs

```diff
@@ -180,20 +202,60 @@ impl FieldPath {
         path
     }
 
+    pub fn new_from_alias(alias: impl Into<String>) -> Self {
+        Self {
+            alias: Some(alias.into()),
+            path: vec![],
+        }
+    }
+
     pub fn add_segment(&mut self, field: &Field) {
         self.path.push(field.db_name().to_owned());
     }
 
-    pub fn path(&self) -> String {
-        self.path.join(".")
+    /// Keep only the last element of the path
+    pub fn keep_only_target(&mut self) {
```

Comment:
`target` implies too much meaning already, `keep_last` is more generic and easily understandable.